src/main/java/org/myrobotlab/opencv/ImageFileFrameGrabber.java

```diff
@@ -12,24 +14,62 @@
 import org.slf4j.Logger;
 
 public class ImageFileFrameGrabber extends FrameGrabber {
-
   public final static Logger log = LoggerFactory.getLogger(ImageFileFrameGrabber.class.getCanonicalName());
 
-  transient private IplImage image;
+  // delay in ms between grabs.
+  // public int delay = 31; // 32 fps is 31.5 ms per frame
+  public int delay = 16; 
+  private ArrayList<File> imageFiles = new ArrayList<File>();
+  private int grabCount = 0;
+  
+  protected transient IplImage image;
   transient private IplImage lastImage;
+  
   transient private HashMap<String, IplImage> cache = new HashMap<String, IplImage>();
   private int frameCounter = 0;
   public String path;
   transient OpenCVFrameConverter.ToIplImage converter = new OpenCVFrameConverter.ToIplImage();
+  
 
   public ImageFileFrameGrabber(String path) {
     this.path = path;
+    load();    
+  }
+
+  public void load() {
+    File target = new File(path);
+    if (!target.isDirectory()) {
+      imageFiles.add(target);
+    } else {
+    File[] listOfFiles = target.listFiles();
+    for (File file : listOfFiles) {
+      if (file.isFile()) {
+        // TODO: check what formats opencv's cvLoadImage supports and add that
+        // here.
+        if (file.getName().toLowerCase().endsWith("png") || file.getName().toLowerCase().endsWith("jpg")) {
+          // It's an image file! ish...
+          imageFiles.add(file);
+        }
+      }
+    }
+  }
   }
 
   @Override
   public Frame grab() {
+    /* not needed as opencv can self regulat - or adjust fps
+    try {
+      // pause for the specified delay before loading the image.
+      Thread.sleep(delay);
+    } catch (InterruptedException e) {}
+    // set the file path
+     * 
+     */
+    path = imageFiles.get(grabCount).getAbsolutePath();
+    log.debug("Grabbing file {} - {}", grabCount, path);
+    // grab it.
     if (!cache.containsKey(path)) {

```

Comment:
This cache is problematic .. if you try to process a large directory with lots of image files, this cache will run us out of memory.  We should re-design this.  (perhaps not for this pull request.. but.. this is definitely not good as it can cause unbounded memory usage.)

Reply:
yup .. agree, but this pr did not change the original design .. it has always been this way.
Sure we could make it better - but its currently worky ;)